jobs/generate_hathifile.rb

```diff
@@ -5,82 +5,94 @@
 require "date"
 require "settings"
 require "push_metrics"
+require "zephir_files"
 
 class GenerateHathifile
-  attr_reader :intype
+  attr_reader :tracker
 
-  def initialize(intype)
-    @intype = intype
+  def initialize
+    @tracker = PushMetrics.new(batch_size: 10_000, job_name: "generate_hathifiles")
   end
 
   def run
-    tracker = PushMetrics.new(batch_size: 10_000, job_name: "generate_hathifile_#{intype}")
-    infile = Dir.glob(File.join(Settings.zephir_dir, "zephir_#{intype}*")).max
+    zephir_files = ZephirFiles.new(
+      zephir_dir: Settings.zephir_dir,
+      hathifiles_dir: Settings.hathifiles_dir
+    )
+    zephir_files.unprocessed.each do |zephir_file|
+      run_file zephir_file
+    end
+    tracker.log_final_line
+  end
 
+  def run_file(zephir_file)
+    infile = File.join(Settings.zephir_dir, zephir_file.filename)
     fin = if /\.gz$/.match?(infile)
       Zlib::GzipReader.open(infile)
     else
       File.open(infile)
     end
 
     # we only want to write some of the items in the zephr records
-    indate = File.basename(infile).split("_")[2].split(".").first
-    cutoff = if /upd/.match?(infile)
-      # cutoff = Date.today.prev_day.strftime("%Y%m%d").to_i
-      indate.to_i - 1
+    cutoff = if zephir_file.type == "upd"
+      zephir_file.date.strftime("%Y%m%d").to_i
```

Comment:
This changes the logic to set cutoff to the date in the zephir file name rather than the previous day, which I believe causes no output to be generated in the typical case. I need to look more at how it's being used.